pkg/api/defaults.go

```diff
@@ -538,6 +544,12 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool, cloudName
 			profile.OSType = Linux
 		}
 
+		// Update fault domain value to 3 for Azure Stack
+		if p.IsAzureStackCloud() && profile.PlatformFaultDomainCount == nil {
```

Comment:
could you please explain me why do we need check the nil? 
in my mind, as long as it is for azure stack, we will set the fault domain count.

Reply:
This is needed to support upgrade of old deployments. Since we are in preview we can the check.